errutil.go

```diff
@@ -63,17 +64,29 @@ type Tags map[string]any
 
 // Wrap enriches err with stack information from the caller of Wrap.
 // The returned error will implement Unwrap (used by errors.Is/As)
-// returning err.
+// returning err. Unwrap will only work for allowed, or all if not given.
 //
 // Generally, Wrap should only be used when the returned error is
 // expected to be used with errors.Is or similar.
-func Wrap(err error) error {
-	return NewFrameError(Caller(1), nil, err, true)
+func Wrap(err error, allowed ...error) error {
+	return wrap(err, nil, allowed...)
 }
 
 // Wrapt is like Wrap but adds tags t to the error information.
-func Wrapt(err error, t Tags) error {
-	return NewFrameError(Caller(1), t, err, true)
+func Wrapt(err error, t Tags, allowed ...error) error {
+	return wrap(err, t, allowed...)
+}
+
+func wrap(err error, t Tags, allowed ...error) error {
+	if len(allowed) == 0 {
+		return NewFrameError(Caller(2), t, err, true)
+	}
+	for _, a := range allowed {
+		if errors.Is(err, a) {
+			return NewFrameError(Caller(2), t, err, true)
+		}
+	}
+	return NewFrameError(Caller(2), t, err, false)
 }
 
```

Comment:
Hm, even on master getting:
```
--- FAIL: TestBuildStack (0.00s)
    --- FAIL: TestBuildStack/AnonymousFunc (0.00s)
        errutil_test.go:162:   errutil.Stack{
              	{
            - 		Pkg:    "errutil/test/dot.pkg",
            + 		Pkg:    "errutil_test",
            - 		Func:   "AnonymousFunc.func1.1",
            + 		Func:   "TestBuildStack.AnonymousFunc.func2.1",
              		File:   "dot.pkg.go",
              		Line:   33,
              		Values: {"k1": bool(true)},
              	},
              }
```

Reply:
If I up the frames from 4 to 6 and then output all of the `runtime.Frame.Function`s I get:
```
github.com/graxinc/errutil_test.TestBuildStack.AnonymousFunc.func2.1
github.com/graxinc/errutil_test.TestBuildStack.AnonymousFunc.func2
github.com/graxinc/errutil/test/dot%2epkg.AnonymousFunc
github.com/graxinc/errutil_test.TestBuildStack
testing.tRunner
runtime.goexit
```
which looks wrong.

Reply:
Actually the `TestBuildStack.AnonymousFunc.func2.1` looks right, since we inlined `dot.pkg.AnonymousFunc`. If I check if any `runtime.Frame.Func` is nil (which is documented to be if inlined) I only get one, which is the one I just mentioned. So I guess this is as good as we can get!

Reply:
Played with a few things that might force an inlining vs not, but didn't see anything consistent to leave in the test.